migration/v0.40/prepare/20-update-google-dex-config.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+HERE="$(dirname "$(readlink -f "${0}")")"
+ROOT="$(readlink -f "${HERE}/../../../")"
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+output=$(sops -d --extract '["dex"]' "${CK8S_CONFIG_PATH}/secrets.yaml" | sed "s/^\([ \t]*\)adminEmail: \(.*\)$/\1domainToAdminEmail: \n\1  \'\*\': \2/" | yq4 '.' -o json)
+
+sops --set "[\"dex\"] ${output}" "${CK8S_CONFIG_PATH}/secrets.yaml"
```

Comment:
Feels like this ought to be possible with `sops exec-file` but apparently not, weird when it can do in the mode where it invokes an editor on the file.

This preserves comments etc:

```suggestion
EDITOR="yq4 -i 'with(.dex.connectors[]; .domainToAdminEmail[\"*\"] = .config.adminEmail | del(.config.adminEmail))'" sops "${CK8S_CONFIG_PATH}/secrets.yaml"
```

Reply:
Updated with the suggestion from you and @aarnq